### ISSUE TITLE
Key error when adding genes to geneDict in prepDE.py

### ISSUE THREAD

**Issue by astrophys** [gpertea/stringtie#232]
When I run `python prepDE.py -i gtflist.txt` using my gtf files that I've generated I get the following : 

```
$ python /opt/stringtie/2.0/prepDE.py -i gtflist.txt
0 BMDM_Pde3A_KO_CGS_M6_S11_L001_R1_001_trimmed
1 BMDM_Pde3A_KO_CGS_M6_S11_L002_R1_001_trimmed
Traceback (most recent call last):
  File "/opt/stringtie/2.0/prepDE.py", line 255, in <module>
    geneDict.setdefault(geneIDs[i],{}) #gene_id
KeyError: 'ENSMUST00000155445'
```
This looks to be identical to the issue identified [Issue #229 ](https://github.com/gpertea/stringtie/issues/229).

My `gtflist.txt` looks like :
```
$ head -n 3 gtflist.txt
BMDM_Pde3A_WT__CGS_M2_S3_L001_R1_001_trimmed     output_ensembl/stringtie-est/star/fr-unstranded/BMDM_Pde3A_WT__CGS_M2_S3_L001_R1_001_trimmed/est.gtf
BMDM_Pde3B_KO_DMSO_M13_S10_L001_R1_001_trimmed   output_ensembl/stringtie-est/star/fr-unstranded/BMDM_Pde3B_KO_DMSO_M13_S10_L001_R1_001_trimmed/est.gtf
BMDM_Pde3A_WT_DMSO_M3_S5_L002_R1_001_trimmed     output_ensembl/stringtie-est/star/fr-unstranded/BMDM_Pde3A_WT_DMSO_M3_S5_L002_R1_001_trimmed/est.gtf
```

I believe the issue comes about when getting bad genes (lines 120-158 in `prepDE.py`).  When looping through `samples`, the `break` statement in the `try except else` section short-circuits the for loop  after reading in only 1 file.  So if there is a gene not annotated in the first gtf file (see `samples` on line 119), then on line 255 you get the above error because you never added any of the transcripts from the other gtf files.

**Question** 
Is this analysis correct, or am I running this code incorrectly.  If it is correct, is the `break` statement on line 158 a bug?

**Comment by AishMandya** [comment]
Didi you find a solution to this ?

**Comment by SofiaZhangtj** [comment]
Hi, I am having the same problem. Do you have a solution to this? May I know how did you solve it, please?

**Comment by astrophys** [comment]
I haven't had time to figure out / debug this issue.

**Comment by gpertea** [comment]
This seems to be the same issue with #234 and #232, so it should be fixed in v2.0.4 release.
